processing/src/main/java/io/druid/segment/DimensionSelector.java

```diff
@@ -42,7 +44,12 @@
    *
    * Value cardinality would be 2.
    *
-   * @return the value cardinality
+   * Cardinality may be unknown (e.g. the selector used by IncrementalIndex while reading input rows),
+   * in which case this method will return -1. If cardinality is unknown, you should assume this
+   * dimension selector has no dictionary, and avoid calling "lookupId" or "lookupName" outside of
```

Comment:
It's probably not too much, let me go back and look at them though. There aren't that many call sites.